tests/storage/test_event_push_actions.py

```diff
@@ -54,154 +62,118 @@ def test_get_unread_push_actions_for_user_in_range_for_email(self) -> None:
-                self.store.db_pool.simple_insert(
-                    table="events",
-                    values={
-                        "stream_ordering": stream,
-                        "topological_ordering": stream,
-                        "type": "m.room.message",
-                        "room_id": room_id,
-                        "processed": True,
-                        "outlier": False,
-                        "event_id": event.event_id,
-                    },
-                )
+        def _inject_actions(highlight: bool = False) -> None:
+            result = self.helper.send_event(
+                room_id,
+                type="m.room.message",
+                content={"msgtype": "m.text", "body": user_id if highlight else ""},
+                tok=other_token,
             )
+            nonlocal last_event_id
+            last_event_id = result["event_id"]
+            return last_event_id
 
-            self.get_success(
-                self.store.add_push_actions_to_staging(
-                    event.event_id,
-                    {user_id: action},
-                    False,
-                )
-            )
-            self.get_success(
-                self.store.db_pool.runInteraction(
-                    "",
-                    self.persist_events_store._set_push_actions_for_event_and_users_txn,
-                    [(event, None)],
-                    [(event, None)],
-                )
-            )
-
-        def _rotate(stream: int) -> None:
-            self.get_success(
-                self.store.db_pool.runInteraction(
-                    "rotate-receipts", self.store._handle_new_receipts_for_notifs_txn
-                )
-            )
-
-            self.get_success(
-                self.store.db_pool.runInteraction(
-                    "rotate-notifs", self.store._rotate_notifs_before_txn, stream
-                )
-            )
-
-        def _mark_read(stream: int, depth: int) -> None:
-            last_read_stream_ordering[0] = stream
+        def _rotate() -> None:
+            self.get_success(self.store._rotate_notifs())
 
+        def _mark_read(event_id) -> None:
             self.get_success(
                 self.store.insert_receipt(
                     room_id,
                     "m.read",
                     user_id=user_id,
-                    event_ids=[f"$test{stream}:example.com"],
+                    event_ids=[event_id],
                     data={},
                 )
             )
 
-        _assert_counts(0, 0)
-        _inject_actions(1, PlAIN_NOTIF)
-        _assert_counts(1, 0)
-        _rotate(1)
-        _assert_counts(1, 0)
+        _assert_counts(0, 0, 0)
+        _inject_actions()
+        _assert_counts(1, 0, 0)
+        _rotate()
+        _assert_counts(1, 0, 0)
 
-        _inject_actions(3, PlAIN_NOTIF)
-        _assert_counts(2, 0)
-        _rotate(3)
-        _assert_counts(2, 0)
+        event_id = _inject_actions()
```

Comment:
In a couple of spots we rotate (or mark as read) not from the latest event, so we need to store another temporary event ID.